### PR TITLE
Registry: read hexadecimal values correctly

### PR DESCRIPTION
**Pull request by Lgt2x**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

Hex string values from the registry containing an alpha character were parsed incorrectly. Use more standard hex parsing instead of hand-made incorrect function

### Related Issues

#590

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Comment by Lgt2x** [comment]
@ccfly42 this should fix PPS reading from #590 :)

File: linux/registry.cpp [hunk 1]
```diff
@@ -60,19 +60,11 @@
...
-    else if ((*p >= 'A') && (*p <= 'F'))
-      value = (value * 16) + ((*p) - 'A');
-    else
-      return value;
-
-    p++;
-  }
+int hextoi(char *hexstring) {
+  int value;
+  std::stringstream ss;
```
Comment (winterheart) [inline linux/registry.cpp:65]: Need `#include <sstream>` for `std::stringstream`.

Comment (Lgt2x) [reply linux/registry.cpp:66]: fixed

**Comment by winterheart** [comment]
Tested on Linux, runs fine.

**Pull request merged**